httpapi/httpapi_test.go

```diff
@@ -306,223 +306,125 @@ func TestDeleteFunction_OK(t *testing.T) {
+		Type:       subscription.TypeSync,
+		EventType:  "http.request",
+		FunctionID: "func",
+		Method:     "GET",
+		Path:       "/",
+		CORS: &subscription.CORS{
+			Origins:          []string{"*"},
+			Methods:          []string{"HEAD", "GET", "POST"},
+			Headers:          []string{"Origin", "Accept", "Content-Type"},
+			AllowCredentials: false,
 		},
 	}
-	subscriptions.EXPECT().UpdateSubscription(subscription.ID("http,GET,%2F"), input).Return(nil, &subscription.ErrSubscriptionValidation{Message: "" })
-
-	resp := httptest.NewRecorder()
-	payload := bytes.NewReader([]byte(`
-	    {"space":"default","subscriptionId":"http,GET,%2F","functionId":"func","method":"GET","path":"/","cors":{"origins":["*"],"methods":["HEAD","GET","POST"],"headers":["Origin","Accept","Content-Type"],"allowCredentials":false}}
-		`))
-	req, _ := http.NewRequest(http.MethodPut, "/v1/spaces/default/subscriptions/http,GET,%2F", payload)
-	router.ServeHTTP(resp, req)
-
-	httpresp := &httpapi.Response{}
-	json.Unmarshal(resp.Body.Bytes(), httpresp)
-	assert.Equal(t, http.StatusBadRequest, resp.Code)
-	assert.Contains(t, httpresp.Errors[0].Message, "Subscription doesn't validate. Validation error")
-}
-
-func TestUpdateSubscription_InternalError(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
-	router, _, subscriptions := setup(ctrl)
-
-	input := &subscription.Subscription{
-		Space:        "default",
-		ID:           subscription.ID("http,GET,%2F"),
-		Event:        "http",
-		FunctionID:   "func",
-		Method:       "GET",
-		Path:         "/",
-		CORS:         &subscription.CORS{
-		    Origins:          []string{"*"},
-		    Methods:          []string{"HEAD", "GET", "POST"},
-		    Headers:          []string{"Origin", "Accept", "Content-Type"},
-		    AllowCredentials: false,
-		},
-	}
-	subscriptions.EXPECT().UpdateSubscription(subscription.ID("http,GET,%2F"), input).Return(nil, errors.New("processing failed"))
-
-	resp := httptest.NewRecorder()
-	payload := bytes.NewReader([]byte(`
-	    {"space":"default","subscriptionId":"http,GET,%2F","event":"http","functionId":"func","method":"GET","path":"/","cors":{"origins":["*"],"methods":["HEAD","GET","POST"],"headers":["Origin","Accept","Content-Type"],"allowCredentials":false}}
-		`))
-	req, _ := http.NewRequest(http.MethodPut, "/v1/spaces/default/subscriptions/http,GET,%2F", payload)
-	router.ServeHTTP(resp, req)
-
-	httpresp := &httpapi.Response{}
-	json.Unmarshal(resp.Body.Bytes(), httpresp)
-	assert.Equal(t, http.StatusInternalServerError, resp.Code)
-	assert.Equal(t, "processing failed", httpresp.Errors[0].Message)
+	updatedValue := []byte(`{"space":"default","subscriptionId":"testid","type":"sync",` +
+		`"eventType":"http.request","functionId":"func","method":"GET","path":"/",` +
+		`"cors":{"origins":["*"],"methods":["HEAD","GET","POST"],` +
+		`"headers":["Origin","Accept","Content-Type"],"allowCredentials":false}}`)
+
+	t.Run("subscription updated", func(t *testing.T) {
```

Comment:
I like this new approach for unit tests. It enabled having setup and teardown steps.